cvmfs/catalog_mgr_rw.cc

```diff
@@ -913,30 +913,42 @@ void WritableCatalogManager::SwapNestedCatalog(const string &mountpoint,
           nested_root_path.c_str(), parent_path.c_str());
   }
 
-  // Get old catalog hash
-  shash::Any old_hash;
-  uint64_t old_size;
-  const bool old_found = parent->FindNested(nested_root_ps, &old_hash,
-                                            &old_size);
-  if (!old_found) {
-    PANIC(kLogStderr,
-          "failed to swap nested catalog '%s': not found in parent",
-          nested_root_path.c_str());
-  }
+  // Get old nested catalog counters
+  Catalog *old_attached_catalog = parent->FindChild(nested_root_ps);
+  Counters old_counters;
+  if (old_attached_catalog) {
```

Comment:
Perhaps one more thing: could you add the example from the PR description as a comment so that we remember why it might be possible that the catalog is already attached.

Reply:
Have done